week4/SignupLogin/src/components/UI/Input.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useRef } from 'react';
+import styled from 'styled-components';
+
+const Input = (props) => {
+  return (
+    <InputDiv>
+      <InputLabel>{props.label}</InputLabel>
+      <UserInput
+        type="text"
+        name={props.label}
+        placeholder={props.placeholder}
+        content={props.content}
+        onChange={props.onChange}
+      ></UserInput>
+      {props.content === '중복체크' && (
+        <IsExistIdBtn onClick={props.onClick} $isDuplicate={props.isDuplicate} $isBtnClicked={props.btnClicked}>
+          {props.content}
+        </IsExistIdBtn>
+      )}
+    </InputDiv>
+  );
+};
```

Comment:
사용자의 입력을 받는 부분 공통 컴포넌트 분리 너무 잘해주셔쎈요!! 최고 ㅎㅎ 
여기서 한단계만 더 깔끔한 코드를 만들어보자면, 현재 `props.` 가 반복되고 있는데, 이 반복되는 객체 접근은 **구조분해할당** 한번만으로 싹 생략시킬 수 있다는 점! 

아래와 같이 바꿔줄 수 있겠죠? ㅎㅎ

전달받은 값에 따라 조건부 렌더링 해준 것도 너무너무 좋아요!! 👍 
```suggestion
const Input = ({label, placeholder, content, onChange, onClick, isDuplicate, btnClicked}) => {
  return (
    <InputDiv>
      <InputLabel>{label}</InputLabel>
      <UserInput
        type="text"
        name={label}
        placeholder={placeholder}
        content={content}
        onChange={onChange}
      ></UserInput>
      {content === '중복체크' && (
        <IsExistIdBtn onClick={onClick} $isDuplicate={isDuplicate} $isBtnClicked={btnClicked}>
          {content}
        </IsExistIdBtn>
      )}
    </InputDiv>
  );
};
```